azure_functions_worker/loader.py

```diff
@@ -91,3 +93,32 @@ def load_function(name: str, directory: str, script_file: str,
             f'present in {rel_script_path}')
 
     return func
+
+
+@attach_message_to_exception(
+    expt_type=ImportError,
+    message=f'Troubleshooting Guide: {MODULE_NOT_FOUND_TS_URL}'
+)
+def index_function_app(directory: str) -> typing.List[Function]:
+    function_path = os.path.join(directory, SCRIPT_FILE_NAME)
+
+    if not os.path.exists(function_path):
+        raise IOError(f"{SCRIPT_FILE_NAME} not found in the function app")
+
+    module_name = pathlib.Path(function_path).stem
+    imported_module = importlib.import_module(module_name)
+
+    app: typing.Optional[FunctionsApp] = None
+    for i in imported_module.__dir__():
+        if isinstance(getattr(imported_module, i, None), FunctionsApp):
+            if not app:
+                app = getattr(imported_module, i, None)
+            else:
+                raise ValueError("Multiple Apps defined")
+
+    if app:
```

Comment:
nit: recommend guarded clause
```
if not app:
    raise ValueError("No Apps defined")

return app.get_functions()
```

Reply:
Again - make the errors very verbose and pointers to how they could fix it (if the documentation page isn't ready)